File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
import sys


def input():
    return sys.stdin.readline().rstrip()


data = input()
stack = []
cnt = 0
for i in range(len(data)):
    if data[i] == '(':
        stack.append('(')
    elif data[i - 1] == '(':
        stack.pop()
        cnt += len(stack)
    else:
        stack.pop()
        cnt += 1
print(cnt)

File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include <iostream>
#include <stack>
#define MAX 100'001
using namespace std;

int main() {
	cin.tie(0)->sync_with_stdio(0);
	
	char s[MAX];
	cin >> s;
	stack<char> st;
	int cnt = 0;
	for (int i = 0; s[i] != '\0'; i++) {
		if (s[i] == '(') st.push(s[i]);
		else if (s[i - 1] == '(') {
			st.pop();
			cnt += st.size();
		}
		else {
			st.pop();
			cnt++;
		}
	}
	cout << cnt;
}